modules/flowable-cmmn-engine/src/main/java/org/flowable/cmmn/engine/impl/runtime/CaseInstanceHelperImpl.java

```diff
@@ -421,6 +472,178 @@ protected CaseInstanceEntity createCaseInstanceEntityFromDefinition(CommandConte
+
+        CaseInstanceEntity caseInstanceEntity = caseInstanceEntityManager.create();
+
+        // we re-use the same id as in the history as we want them to still be connected, the history will be updated from the reactivated case as we move on
+        caseInstanceEntity.setId(historicCaseInstance.getId());
+
+        // set the state back to active, not copying it from the historic one, obviously
+        caseInstanceEntity.setState(CaseInstanceState.ACTIVE);
+
+        caseInstanceEntity.setBusinessKey(historicCaseInstance.getBusinessKey());
+        caseInstanceEntity.setName(historicCaseInstance.getName());
+        caseInstanceEntity.setParentId(historicCaseInstance.getParentId());
+        caseInstanceEntity.setCaseDefinitionId(historicCaseInstance.getCaseDefinitionId());
+        caseInstanceEntity.setCaseDefinitionKey(historicCaseInstance.getCaseDefinitionKey());
+        caseInstanceEntity.setCaseDefinitionName(historicCaseInstance.getCaseDefinitionName());
+        caseInstanceEntity.setCaseDefinitionVersion(historicCaseInstance.getCaseDefinitionVersion());
+        caseInstanceEntity.setCaseDefinitionDeploymentId(historicCaseInstance.getCaseDefinitionDeploymentId());
+        caseInstanceEntity.setStartTime(historicCaseInstance.getStartTime());
+        caseInstanceEntity.setStartUserId(historicCaseInstance.getStartUserId());
+        caseInstanceEntity.setCallbackId(historicCaseInstance.getCallbackId());
+        caseInstanceEntity.setCallbackType(historicCaseInstance.getCallbackType());
+        caseInstanceEntity.setReferenceId(historicCaseInstance.getReferenceId());
+        caseInstanceEntity.setReferenceType(historicCaseInstance.getReferenceType());
+        caseInstanceEntity.setTenantId(historicCaseInstance.getTenantId());
+
+        // we need the case id being set before creating the plan items and variables associated with the case instance, so we need to insert it at this point
+        caseInstanceEntityManager.insert(caseInstanceEntity);
+
+        // now copy the variables from the history to the runtime as well and set them on the entity to make sure further processing (e.g. evaluating expressions)
+        // will actually work
+        if (caseInstanceEntity instanceof CaseInstanceEntityImpl) {
+            ((CaseInstanceEntityImpl) caseInstanceEntity)
+                .setVariableInstancesToCache(createCaseVariablesFromHistoricCaseInstance(historicCaseInstance, caseInstanceEntity));
+        } else {
+            throw new FlowableIllegalArgumentException("Cannot set variables on reactivated case instance as it is not from expected implementation class.");
+        }
+
+        // create runtime plan items from the history and set them as the new child plan item list
+        caseInstanceEntity.setChildPlanItemInstances(createCasePlanItemsFromHistoricCaseInstance(historicCaseInstance, caseInstanceEntity));
+
```

Comment:
trigger history manager reactivate method here -> and move the logic below there